packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1525,6 +1524,34 @@ export class ContainerRuntime
 			opGroupingManager,
 		);
 
+		const deltaManagerSummarizerProxy = new DeltaManagerSummarizerProxy(this.innerDeltaManager);
+		const pendingRuntimeState = pendingLocalState as IPendingRuntimeState | undefined;
+		this.pendingStateManager = new PendingStateManager(
+			{
+				applyStashedOp: this.applyStashedOp.bind(this),
+				clientId: () => this.clientId,
+				close: this.closeFn,
+				connected: () => this.connected,
+				reSubmit: (message: IPendingBatchMessage) => {
+					this.reSubmit(message);
+					this.flush();
+				},
+				reSubmitBatch: this.reSubmitBatch.bind(this),
+				isActiveConnection: () => this.innerDeltaManager.active,
+				isAttached: () => this.attachState !== AttachState.Detached,
+			},
+			pendingRuntimeState?.pending,
+			this.logger,
+		);
+
+		const useDeltaManagerOpsProxy =
+			this.mc.config.getBoolean("Fluid.ContainerRuntime.DeltaManagerOpsProxy") === true;
+		// The DeltaManagerPendingOpsProxy is used to control the minimum sequence number
+		// It allows us to lie to the layers below so that they can maintain enough local state for rebasing ops.
+		this.deltaManager = useDeltaManagerOpsProxy
```

Comment:
minor: in order for pattern to be repeatable, I'd rather see something like that.

```ts
let innerManager = new DeltaManagerSummarizerProxy(this.innerDeltaManager);
if (useDeltaManagerOpsProxy) {
    innerManager = new DeltaManagerPendingOpsProxy(deltaManagerSummarizerProxy, this.pendingStateManager);
}
this.deltaManager = innerManager;
```

This way if we need to add another optional wrapper in the future, it's easier to do

Reply:
Updated